var/spack/repos/builtin/packages/cp2k/package.py

```diff
@@ -164,19 +164,15 @@ def install(self, spec, prefix):
                         'libmetis.{0}'.format(dso_suffix)
                     ),
                 ])
-                libs.extend(spec['scalapack'].scalapack_shared_libs)
+                libs.extend(scalapack)
                 libs.extend(self.spec['mpi'].mpicxx_shared_libs)
                 libs.extend(self.compiler.stdcxx_libs)
             # LAPACK / BLAS
-            ldflags.extend([
-                '-L' + spec['lapack'].prefix.lib,
-                '-L' + spec['blas'].prefix.lib
-            ])
-            libs.extend([
-                join_path(spec['fftw'].prefix.lib, 'libfftw3.so'),
-                spec['lapack'].lapack_shared_lib,
-                spec['blas'].blas_shared_lib
-            ])
+            lapack = spec['lapack'].lapack_libs
+            blas = spec['blas'].blas_libs
+
+            ldflags.append(lapack.search_flags + blas.search_flags)
```

Comment:
Well spotted! The issue should be fixed now. Maybe @KineticTheory would like to have a look before merging at netlib-scalapack.